wp/lib/bap_wp/src/precondition.ml

```diff
@@ -467,6 +488,13 @@ let rec get_vars (env : Env.t) (t : Sub.t) : Var.Set.t =
   in
   visitor#visit_sub t vars
 
+let get_vars (env : Env.t) (t : Sub.t) : Var.Set.t =
+  let gprs = Env.get_gprs env in
+  let mem = Var.Set.singleton (Env.get_mem env) in
+  let sp = Var.Set.singleton (Env.get_sp env) in
+  let sub_vars = vars_from_sub env t in
+  Var.Set.union_list [gprs; mem; sp; sub_vars]
```

Comment:
Didn't realize that function existed! It might be a good idea to return a union of the `sub_vars` and the `Target.vars` to capture everything including the virtual registers. I can test this out.

Reply:
Nah, that would be redundant. If we can use just `Target.vars` then we should, but I'm not sure it's available/doesn't break backwards compatibility.

Unioning both is ugly and will probably be problematic at some point.

Reply:
I tried testing it out. It breaks backwards compatibility since none of wp's library uses the core theory. Doing this would probably require a big rewrite.

Reply:
Thought it might. Let's drop it for the time being.